src/analyzer/analyzerkey.cpp

```diff
@@ -155,12 +157,42 @@ bool AnalyzerKey::processSamples(const CSAMPLE* pIn, SINT count) {
         return false;
     }
 
-    m_currentFrame += count / mixxx::kAnalysisChannels;
+    SINT numFrames = count / m_channelCount;
+    m_currentFrame += numFrames;
+
     if (m_currentFrame > m_maxFramesToProcess) {
         return true; // silently ignore remaining samples
     }
 
-    return m_pPlugin->processSamples(pIn, count);
+    const CSAMPLE* pKeyInput = pIn;
+    CSAMPLE* pHarmonicMixedChannel = nullptr;
+
+    if (m_channelCount > mixxx::audio::ChannelCount::stereo()) {
+        // If we have multi channel file (a stem file), we mix all the stems
+        // together except the first one which contains drums or beats by
+        // convention
+        // Note that if Mixxx adds support for other multi channel sound sources
+        // (e.g 5:1 dolby), this condition will have to be reworked to assert if
+        // the sound source is a NI stem
```

Comment:
> This would also fix also the 6ch Dolby 5:1 case

It doesn't since some of those analysers relies on stereo signals. 
Since Dolby 5:1 handling seems to have become a requirement for that feature, I have added an assert and an early return.